# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/abort/AbortFollower.h
# Commit: d27c32ae0908
# Full Hash: d27c32ae0908c0b5d2c9213b9a624e3d0b71e2d5
# Author: Vincent Hilla <vhilla@mozilla.com>
# Date: 2024-01-23 21:57:50
# Regressor Bug: 1830781
# File Overlap Count: 1
# Description:
#   Bug 1830781 - Implement AbortSignal.any() r=saschanaz,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D198011
# ==============================================================================

diff -r 8af7503ecb57 -r d27c32ae0908 dom/abort/AbortFollower.h
--- a/dom/abort/AbortFollower.h	Tue Jan 23 08:04:52 2024 +0000
+++ b/dom/abort/AbortFollower.h	Tue Jan 23 08:43:07 2024 +0000
@@ -44,6 +44,11 @@
   WeakPtr<AbortSignalImpl> mFollowingSignal;
 };
 
+/*
+ * AbortSignalImpl is a minimal implementation without an associated global
+ * and without event dispatching, those are added in AbortSignal.
+ * See Bug 1478101
+ */
 class AbortSignalImpl : public nsISupports, public SupportsWeakPtr {
  public:
   explicit AbortSignalImpl(bool aAborted, JS::Handle<JS::Value> aReason);
@@ -67,6 +72,8 @@
 
   virtual ~AbortSignalImpl() { UnlinkFollowers(); }
 
+  void SetAborted(JS::Handle<JS::Value> aReason);
+
   JS::Heap<JS::Value> mReason;
 
  private: